### DISCUSSION TITLE
[Hackathon Use Case] - FDC3 Data Mesh for Charting

### DISCUSSION THREAD

**Discussion by maoo** [finos/finos-pmcs#243] [Ideas]
# FDC3 Data Mesh for Charting
_From @jonfreedman (Brevan Howard)_

## Business Problem
It's a common problem that discretionary trades want to be able to produce charts using data from a variety of sources, there are various dealer tools that offer this (Barclays BARX, Goldman Sachs PlotTool, JPMorgan DataQuery) but each will only have the data available within the respective bank. In addition most consumers of these tools will have their own datasets, plus those from market data providers. If you want to pull various pieces of data together this typically requires establishing a market data team to source data from the various vendors into a bespoke platform. That platform then requires a variety of access controls to be put in place. If it were possible for a user to do this on the desktop then they would be creating a chart based on data directly accessed from the golden source systems.

## Expected features

- Mock up 2+ source systems, that are able to expose data via FDC3, send data from these into a separate FDC3 charting app.
- Some consideration or implementation given to streaming data not just static
- Functionality to share a chart between users, displaying data if available and warnings about anything missing

## Useful apps, systems, projects, technologies, libraries and datasets

- OpenFin/Glue42 (& possibly Finsemble) for Bloomberg terminal connectivity
- ag-grid to display data in tabular form
- highcharts / d3 / chartiq / perspective to display data
- https://www.highcharts.com/demo/stock/all-indicators

**Comment by robmoffat** [comment]
"Creating charts from multiple data sources"

**Comment by jonfreedman** [comment]
Should read discretionary traders not trades

**Comment by TheJuanAndOnly99** [comment]
Closing discussion in preparation [for this years hackathon](https://www.finos.org/hosted-events/2024-10-25-citi-india-hackathon-supported-by-finos). Feel free to reopen and update if its still relevant.
